utils/vscode/src/extension.ts

```diff
@@ -26,29 +26,60 @@ let client: LanguageClient;
 /**
  * Splits a CLI-style quoted string.
  */
-function splitQuotedString(carbonArgs: string): string[] {
-  const regex = /"([^"]*)"|'([^']*)'|(\S+)/g;
-  const result = [];
-  let match;
+function splitQuotedString(argsString: string): string[] {
+  const args: string[] = [];
+  let arg = '';
+  let inSingleQuotes = false;
+  let inDoubleQuotes = false;
+  let escaped = false;
 
-  while ((match = regex.exec(carbonArgs)) !== null) {
-    if (match[1]) {
-      result.push(match[1]);
-    } else if (match[2]) {
-      result.push(match[2]);
-    } else if (match[3]) {
-      result.push(match[3]);
+  for (const char of argsString) {
+    if (escaped) {
+      arg += char;
+      escaped = false;
+      continue;
     }
+    switch (char) {
+      case '\\':
+        escaped = true;
+        continue;
+      case "'":
+        if (!inDoubleQuotes) {
+          inSingleQuotes = !inSingleQuotes;
+          continue;
+        }
+        break;
+      case '"':
+        if (!inSingleQuotes) {
+          inDoubleQuotes = !inDoubleQuotes;
+          continue;
+        }
+        break;
+      case ' ':
+        if (!inSingleQuotes && !inDoubleQuotes) {
+          args.push(arg);
+          arg = '';
+        }
+        break;
+    }
+    arg += char;
   }
-  return result;
+
+  if (arg.length > 0) {
+    args.push(arg);
+  }
+
+  return args;
 }
```

Comment:
```suggestion
function splitQuotedString(argsString: string): string[] {
  const args: string[] = [];
  let arg = '';
  let inSingleQuotes = false;
  let inDoubleQuotes = false;
  let escaped = false;
  let empty = true;

  for (const char of argsString) {
    empty = false;
    if (escaped) {
      arg += char;
      escaped = false;
      continue;
    }
    switch (char) {
      case '\\':
        escaped = true;
        continue;
      case "'":
        if (!inDoubleQuotes) {
          inSingleQuotes = !inSingleQuotes;
          continue;
        }
        break;
      case '"':
        if (!inSingleQuotes) {
          inDoubleQuotes = !inDoubleQuotes;
          continue;
        }
        break;
      case ' ':
        if (!inSingleQuotes && !inDoubleQuotes) {
          if (!empty) {
            args.push(arg);
            arg = '';
            empty = true;
          }
          continue;
        }
        break;
    }
    arg += char;
  }

  if (!empty) {
    args.push(arg);
  }

  return args;
}
```
Some tweaks to handle `""` as an argument, multiple spaces between arguments, and an argument list ending in a space.

Reply:
Adopting this, but in the future I'd find smaller deltas more helpful. [The delta is broken](https://github.com/carbon-language/carbon-lang/pull/5056/commits/5c988f76b69daac084c826eae8a93d1ca79f9e0b) and it's difficult to actually understand the changes here without applying them.

Reply:
Note, part of the issue here I think is you're making a suggestion around code that I assume GitHub was not happy to take a suggestion for.

Reply:
Yeah, github's suggestion feature is a bit broken, unfortunately. I'll try to split the suggestion up into chunks in future.